curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/record-collection.english.md

```diff
@@ -30,20 +30,22 @@ You may refer back to <a href="learn/javascript-algorithms-and-data-structures/b
 
 ```yml
 tests:
+  - text: Do not modify the <code>collection</code> object
+    testString: assert(code.match(/var collection = {\s*2548[:] {\s*"album"[:] "Slippery When Wet",\s*"artist"[:] "Bon Jovi",\s*"tracks"[:] \[\s*"Let It Rock",\s*"You Give Love a Bad Name"\s*\]\s*},\s*2468[:] {\s*"album"[:] "1999",\s*"artist"[:] "Prince",\s*"tracks"[:] \[\s*"1999",\s*"Little Red Corvette"\s*\]\s*},\s*1245[:] {\s*"artist"[:] "Robert Palmer",\s*"tracks"[:] \[ \]\s*},\s*5439[:] {\s*"album"[:] "ABBA Gold"\s*}\s*};/g));
```

Comment:
What you have here basically works and I like the direction you took.  However,  I think it might be more maintainable to just add the following to the After Test section:
```
var collectionCopy = {
  2548: {
    "album": "Slippery When Wet",
    "artist": "Bon Jovi",
    "tracks": [
      "Let It Rock",
      "You Give Love a Bad Name"
    ]
  },
  2468: {
    "album": "1999",
    "artist": "Prince",
    "tracks": [
      "1999",
      "Little Red Corvette"
    ]
  },
  1245: {
    "artist": "Robert Palmer",
    "tracks": [ ]
  },
  5439: {
    "album": "ABBA Gold"
  }
};
```

Then, the testString simply becomes:
```suggestion
    testString: assert(JSON.stringify(collectionCopy) === JSON.stringify(collection));
```

The above would avoids the complex regex.

Reply:
Thanks for the review @RandellDawson 

I like the method here - but I'm not sure how to make it work - the function in the seed code modifies the original object when the tests are ran: 
```
// Alter values below to test your code
updateRecords(5439, "artist", "ABBA");
```
So it adds that record to the original object - and the users are supposed to use it to test their code, so they can change it to anything. So I'm not sure how to check with this method without removing that... or something - any ideas?

Also, do you think I should remove the quotes around the property names?

Reply:
@moT01 OK, after much testing, I concur that since the camper can add extra calls to test (for the camper's own purposes) and these would alter the final testing, for now it is best to go with a regex version like you first attempted.  

I was wondering if you would be OK with a simplified version of the `testString` like the following? It is possible now that you have removed the quotation marks from the property names:
```
    testString: assert(code.match(/var collection = {\s*2548: {\s*album: "Slippery When Wet",\s*artist: "Bon Jovi",\s*tracks: \[\s*"Let It Rock",\s*"You Give Love a Bad Name"\s*\]\s*},\s*2468: {\s*album: "1999",\s*artist: "Prince",\s*tracks: \[\s*"1999",\s*"Little Red Corvette"\s*\]\s*},\s*1245: {\s*artist: "Robert Palmer",\s*tracks: \[ \]\s*},\s*5439: {\s*album: "ABBA Gold"\s*}\s*};/g));
```

Reply:
Would you like me to make a suggestion you can commit?